Documentation/app-operations.md

```diff
@@ -0,0 +1,197 @@
+# Fine-Grained Pod Manipulations
+To provide imperative application operations (e.g. start|stop|inject) capabilities for finer-grained control 
+over rkt containerization concepts and debugging needs, this proposal introduces new stage1 entrypoints that 
+will be used for manipulating applications inside pods.
+
+The motivation behind this change is the direction orchestration systems are taking
+and to facilitate this new direction. For more details, see [kubernetes#25899](https://github.com/yujuhong/kubernetes/blob/08dc66113399c89e31f6872f3c638695a6ec6a8d/docs/proposals/container-runtime-interface-v1.md).
+
+# API
+
+The envisioned workflow for the app-level CLI API is that After a pod has been started, 
+the users will be invoking the rkt CLI which is application logic on top of stage1 entrypoints (introduced below).
+
+## rkt app inject
+Injects an application image into a running pod.
+
+```bash
+rkt app inject <pod-uuid> <image-name/hash/address/registry-URL> <arguments>
+```
+
+## rkt app eject
+Ejects an application image into a running pod, meaning that the traces of it are removed from the pod.
+
+```bash
+rkt app eject <pod-uuid> --app=<app-name> <arguments>
+```
+
+**Note:** Currently we don't run garbage collection while the pod is still running, should
+we consider doing it once we introduce ejection?
+
+## rkt app start
+Starts an application that was either stopped or injected.
```

Comment:
What does it mean to start an applicaiton - e.g. does it involve prestart hooks? or just the main app command line?